controllers/keda/hpa.go

```diff
@@ -221,9 +221,19 @@ func (r *ScaledObjectReconciler) getScaledObjectMetricSpecs(ctx context.Context,
 	}
 
 	metricSpecs := cache.GetMetricSpecForScaling(ctx)
+	var mSpecs []autoscalingv2.MetricSpec
+	var cpuUtilizationMetric int
 
 	for _, metricSpec := range metricSpecs {
 		if metricSpec.Resource != nil {
+			target := metricSpec.Resource.Target
```

Comment:
Would be helpful to add a comment on this block to explain why we are doing this magic, and reference the issue